main/options/options.cc

```diff
@@ -994,12 +996,22 @@ void readOptions(Options &opts,
         }
 
         if (raw.count("e") == 0 && opts.inputFileNames.empty() && !raw["version"].as<bool>() && !opts.runLSP &&
-            opts.storeState.empty()) {
+            opts.storeState.empty() && !opts.print.PayloadSources.enabled) {
             logger->error("You must pass either `{}` or at least one folder or ruby file.\n\n{}", "-e",
                           options.help({""}));
             throw EarlyReturnWithCode(1);
         }
 
+        if (opts.print.PayloadSources.enabled) {
+            if (raw.count("e") > 0) {
+                logger->error("You can't pass both `{}` and `{}`.", "--print=payload-sources", "-e");
+                throw EarlyReturnWithCode(1);
+            } else if (!opts.inputFileNames.empty()) {
+                logger->error("You can't pass both `{}` and paths to typecheck.", "--print=payload-sources");
+                throw EarlyReturnWithCode(1);
+            }
+        }
+
```

Comment:
These restrictions seem ~slightly artificial? Does the correctness of the implementation depend on it?

If the user is in a folder with a `sorbet/config` file, it'll have passed `--dir=.` in the config file (usually).

If it's not required for correctness, maybe we can just allow those things to be passed an ignore them?

Reply:
Note for self: We should also reject `--no-stdlib` with this option.

Reply:
makes sense thanks!